src/urllib3/exceptions.py

```diff
@@ -29,9 +30,11 @@ class HTTPWarning(Warning):
 class PoolError(HTTPError):
     """Base exception for errors caused within a pool."""
 
-    pool: "ConnectionPool"
+    pool: Union["ConnectionPool", "SOCKSConnection"]
 
-    def __init__(self, pool: "ConnectionPool", message: str) -> None:
+    def __init__(
+        self, pool: Union["ConnectionPool", "SOCKSConnection"], message: str
```

Comment:
Hmmm, this is unfortunate.

Is there a better error we could be raising from `_new_conn()`? What do regular `HTTPConnection` objects raise in the case of non-timeout failure?

Might be a good time for a breaking change.

Reply:
Regular `HTTPConnection` objects raise `NewConnectionError` too.

Why are we changing `PoolError` instead of just `NewConnectionError`? In both `HTTPConnection._new_conn` and `SOCKSConnection._new_conn` the `pool` parameter is actually `self`, so not a pool object.